README.md

```diff
@@ -1 +1,130 @@
-# earthstar-fetch
\ No newline at end of file
+# earthstar-fetch
+
+A `fetch()` interface for talking to EarthStar
+
+## Goals
+
+- Stick to existing conventions in HTTP-land (body, methods, prefer standard headers)
+- GET data from earthstar workspaces
+- GET an author address for a given name
+- PUT data into a workspace
+- Auto-generate identity keys from a 4 character name
+- Search via querystring params and path prefixes
+
+## Dream API
+
+Basic URL format:
+
+```
+earthstar://+gardening.xxxxxxxxx/path/to/key.json
+```
+
+GET document
+
+```JavaScript
+const res = await fetch('earthstar://+gardening.xxxxx/example')
+
+// Get the timestamp for when it was last updated
+res.headers.get('mtime') === "1231231231231"
+// Get the author for this document
+res.headers.get('earthstar-author') === '@bbbb.xxxxxxxx'
+// Get the signature for this document, in case you want it 🤷
+res.headers.get('earthstar-signature') === 'xxxxxxx'
+
+// Get the content of the document back out.
+// If it's JSON, you should use `.json()`
+const content = await res.text()
+```
+
+GET an identity 'name' - Generate an identity
+
+```JavaScript
+// Specify `@XXXX` where `X` is a lowercase ascii character
+// This will generate and store a keypair for this identity
+const res = fetch('earthstar://@xmpl')
```

Comment:
> generate and store

store... in a browser keychain or something?

(Notes: identity generation is nondeterministic and there will likely be collisions between the 4-letter shortnames, so don't use the shortname as a key for looking up identities)

Reply:
At the moment I was thinking it'd be stored either on disk in say, Agregore and node, and in localStorage or wherever web pages usually store it with Earthstar.

I'm not really sure how browser keychains would work TBH. Electron doesn't expose them in a nice way. 😅

Regarding identity generation, I was thinking that the shortname would get persisted on your computer so when you reuse it across domains it'd be the same one. Two different users would generate different identities for shortnames on their comptuers though. Also I don't think you should be able to reference other people's identities by shortname in the handler. 🤷

Reply:
Ok :D I might grudgingly accept using shortnames to look up your own identities from the browser memory.  It would be convenient.  Definitely not for other people's identities, there will be collisions there.